runtime/util/hshelp.c

```diff
@@ -482,6 +485,173 @@ areSingleSlotConstantRefsIdentical(J9ROMConstantPoolItem * romCP1, U_32 index1,
 	return utfsAreIdentical(J9ROMSTRINGREF_UTF8DATA((J9ROMStringRef *) ref1), J9ROMSTRINGREF_UTF8DATA((J9ROMStringRef *) ref2));
 }
 
+/*
+ * Helper methods for areCallSiteDataMethodsEquivalent
+ */
+
+/* ROM call site data header size (16 bits of method handle address + additional argument count) */
+static const U_16 callSiteHeaderOffset = 2;
+
+static UDATA
+isMethodHandleAField(UDATA cpType) {
+	return (cpType == MH_REF_GETFIELD) || (cpType == MH_REF_GETSTATIC) || (cpType == MH_REF_PUTSTATIC) || (cpType == MH_REF_PUTFIELD);
+}
+
+static U_16*
+findBSMDataAtIndex(U_16* bsmData, U_16 bsmIndex) {
+	U_16 i = 0;
+	for (i = 0; i < bsmIndex; i++) {
+		/* increment by size of bsm data plus header */
+		bsmData += (bsmData[1] + callSiteHeaderOffset);
+	}
+	return bsmData;
+}
+
+static U_32
+iterateToNextArgument(U_32 sigIndex, U_32 sigLength, U_8* sigData) {
+	if (sigIndex >= sigLength) return sigIndex;
+
+	/* check for object */
+	if (sigData[sigIndex] == 'L') {
+		while((sigIndex < sigLength) && (sigData[sigIndex] != ';')) {
+			sigIndex++;
```

Comment:
We've been trying to switch to `sigIndex += 1` pattern rather than `++` as it tends to be clearer.